char type variable compare with EOF(-1) error in build type -Werror=type-limits

_Please use the following template to help us solving your issue._

## Issue Details
compilation error:
     char default as unsigned char in ARM platform, cause char type variable compare with EOF(-1) error in build type -Werror=type-limits

## Source Code
### example:
```
 char c;
  ....
 if (c == EOF)
    printf("CoreIO::read_from_file(), unexpected end of file.");
```

### error log:
```
CGAL/CORE/CoreIO_impl.h:106:9: error: comparison is always false due to limited range of data type [-Werror=type-limits]
   if (c == EOF)

CGAL/CORE/CoreIO_impl.h:194:60: error: comparison is always true due to limited range of data type [-Werror=type-limits]
   for (int i=0; (i<size)&&((c=skip_backslash_new_line(in)) != EOF ); i++) {
```

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits):  Linux, 64bit
* Compiler: aarch64-linux-gnu-g++, aarch64-linux-gnu-gcc
* Release or debug mode: Release
* Specific flags used (if any):  -Werror=type-limits
* CGAL version: 5.0
* Boost version: libboost1.65-dev:amd64
* Other libraries versions if used (Eigen, TBB, etc.):

--- a/CGAL_Core/include/CGAL/CORE/CoreIO_impl.h
+++ b/CGAL_Core/include/CGAL/CORE/CoreIO_impl.h
@@ -103,7 +103,7 @@ int skip_comment_line (std::istream & in) {
     }
   } while (c == ' ' || c == '\t' || c == '\n');
 
-  if (c == EOF)
+  if (in.eof())
     core_io_error_handler("CoreIO::read_from_file()","unexpected end of file.");
 
   in.putback(c);
@@ -191,7 +191,11 @@ void read_base_number(std::istream& in, BigInt& m, long length, long maxBits) {
 
   buffer = new char[size+2];
   // read digits
-  for (int i=0; (i<size)&&((c=skip_backslash_new_line(in)) != EOF ); i++) {
+  for (int i=0; i<size; i++) {
+    c=skip_backslash_new_line(in);
+    if(in.eof()){
+      break;
+    }
     if (c != ' ' && c != '\t' && c != '\n')
       append_char(buffer, size, pos++, c);
   }

